runtime/tr.source/trj9/build/rules/ztpf-gcc/filetypes.mk

```diff
@@ -0,0 +1,235 @@
+#
+# This Source Code may also be made available under the following
+# Secondary Licenses when the conditions for such availability set
+# forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+# General Public License, version 2 with the GNU Classpath
+# Exception [1] and GNU General Public License, version 2 with the
+# OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] http://openjdk.java.net/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+#
+# These are the rules to compile files of type ".x" into object files
+# as well as to generate clean and cleandeps rules
+#
+
+#
+# Compile .c file into .o file
+#
+define DEF_RULE.c
+
+$(1): $(2) $(1)$(DEPSUFF) | jit_createdirs
+	$$(C_CMD) $$(C_FLAGS) $$(patsubst %,-D%,$$(C_DEFINES)) $$(patsubst %,-I'%',$$(C_INCLUDES)) -MMD -MF $$@$$(DEPSUFF) -MT $$@ -MP -o $$@ -c $$<
+
+$(1)$(DEPSUFF): ;
+
+-include $(1)$(DEPSUFF)
+
+JIT_DIR_LIST+=$(dir $(1))
+
+jit_cleanobjs::
+	rm -f $(1)
+
+jit_cleandeps::
+	rm -f $(1)$(DEPSUFF)
+
+endef # DEF_RULE.c
+
+RULE.c=$(eval $(DEF_RULE.c))
+
+#
+# Compile .cpp file into .o file
+#
+define DEF_RULE.cpp
+$(1): $(2) $(1)$(DEPSUFF) | jit_createdirs
+	$$(CXX_CMD) $$(CXX_FLAGS) $$(patsubst %,-D%,$$(CXX_DEFINES)) $$(patsubst %,-I'%',$$(CXX_INCLUDES)) -MMD -MF $$@$$(DEPSUFF) -MT $$@ -MP -o $$@ -c $$<
+
+$(1)$(DEPSUFF): ;
+
+-include $(1)$(DEPSUFF)
+
+JIT_DIR_LIST+=$(dir $(1))
+
+jit_cleanobjs::
+	rm -f $(1)
+
+jit_cleandeps::
+	rm -f $(1)$(DEPSUFF)
+
+endef # DEF_RULE.cpp
+
+RULE.cpp=$(eval $(DEF_RULE.cpp))
+
+#
+# Compile .s file into .o file
+#
+define DEF_RULE.s
+$(1): $(2) | jit_createdirs
+	$$(S_CMD) $$(S_FLAGS) $$(patsubst %,--defsym %=1,$$(S_DEFINES)) $$(patsubst %,-I'%',$$(S_INCLUDES)) -o $$@ $$<
+
+JIT_DIR_LIST+=$(dir $(1))
+
+jit_cleanobjs::
+	rm -f $(1)
+
+endef # DEF_RULE.s
+
+RULE.s=$(eval $(DEF_RULE.s))
+
+##### START X SPECIFIC RULES #####
+ifeq ($(HOST_ARCH),x)
+
+# TODO - Fix MASM2GAS to have a little more sane output file behavior
+#
+# Compile .asm file into .o file
+# (By changing it to a .s file and then assembling it)
+#
+define DEF_RULE.asm
+$(1).s: $(2) | jit_createdirs
+	$$(PERL_PATH) $$(ASM_SCRIPT) $$(ASM_FLAGS) $$(patsubst %,-I'%',$$(ASM_INCLUDES)) -o$$(dir $$@) $$<
+	-mv $$(dir $$@)$$(notdir $$(basename $$<).s) $$@
+
+JIT_DIR_LIST+=$(dir $(1))
+
+jit_cleanobjs::
+	rm -f $(1).s
+
+$(call RULE.s,$(1),$(1).s)
+endef # DEF_RULE.asm
+
+RULE.asm=$(eval $(DEF_RULE.asm))
+
+#
+# Compile .pasm file into .o file
+#
+define DEF_RULE.pasm
+$(1).asm: $(2) | jit_createdirs
+	$$(PASM_CMD) $$(PASM_FLAGS) $$(patsubst %,-I'%',$$(PASM_INCLUDES)) -o $$@ -x assembler-with-cpp -E -P $$<
+
+JIT_DIR_LIST+=$(dir $(1))
+
+jit_cleanobjs::
+	rm -f $(1).asm
+
+$(call RULE.asm,$(1),$(1).asm)
+endef # DEF_RULE.pasm
+
+RULE.pasm=$(eval $(DEF_RULE.pasm))
+
+endif # ($(HOST_ARCH),x)
+##### END X SPECIFIC RULES #####
+
+##### START PPC SPECIFIC RULES #####
```

Comment:
Is this section necessary in a ztpf file?

Reply:
Not necessary...